pynemo/nemo_bdy_extr_tm3.py

```diff
@@ -322,23 +326,34 @@ def __init__(self, setup, SourceCoord, DstCoord, Grid, var_nam, grd, pair):
 #        id_121 = id_121 + reptile * (id_121 == len(dst_lon))
 
         rep_dims = (id_121.shape[0], id_121.shape[1], sc_z_len)
+        rep_dims_2d = (id_121.shape[0], id_121.shape[1], 1)
         # These tran/tiles work like matlab. Tested with same Data.
+        id_121_2d = id_121.repeat(1).reshape(rep_dims_2d).transpose(2, 0, 1)
+        reptile = np.arange(1).repeat(num_bdy).reshape(1, 
+                                                              num_bdy)
+        reptile = reptile.repeat(3).reshape(num_bdy, 3, 1, 
+                                            order='F').transpose(2, 0, 1)
+
+        id_121_2d = sub2ind((1, num_bdy), id_121_2d, reptile)
+
```

Comment:
It looks like in a effort to clean things up I did away with the 'z' grid and grouped all tracers and ssh under the 't' grid - this caused a few issues, so I have to carry around a 2d and 3d version of things. These are all quick fixes to get things working. I will remove the Matlab-style vectorisation in favour of loops and NUMBA in the next iteration.